src/Adapters/TraceListener.Shared/ApplicationInsightsTraceListener.cs

```diff
@@ -47,6 +47,16 @@ public ApplicationInsightsTraceListener(string instrumentationKey)
             this.TelemetryClient.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("sd:");
         }
 
+        /// <summary>
+        /// Initializes a new instance of the ApplicationInsightsTraceListener class, specifying a TelemetryClient to use
+        /// </summary>
+        /// <param name="telemetryClient">TelemetryClient to use for tracing</param>
+        public ApplicationInsightsTraceListener(TelemetryClient telemetryClient)
```

Comment:
Thanks for the reply Sergey! I'm not sure if that would work, my goal is to be able to change the Session.Id during execution. With my change I could do that by modifying the telemetryClient that I'd previously passed into the TraceListener constructor.

I don't see any way to set the session with the TelemetryConfiguration class, so I don't *think* that'd work for me. Please correct me if I'm wrong